lightning-graph-sync/src/lib.rs

```diff
@@ -0,0 +1,241 @@
+use core::ops::Deref;
+use std::fs::File;
+
+use lightning::chain;
+use lightning::routing::network_graph;
+use lightning_block_sync::http::*;
+
+use crate::error::GraphSyncError;
+
+pub mod error;
+pub mod processing;
+
+pub async fn sync_network_graph_with_url<CS: Deref>(
+    network_graph: &network_graph::NetworkGraph,
```

Comment:
Can we use tabs here too?

Reply:
Done (I created an .editorconfig locally, we might wanna have one in the project tree)